horovod/common/ops/nccl_operations.cc

```diff
@@ -43,6 +43,12 @@ void NCCLContext::ErrorCheck(std::string op_name, ncclResult_t nccl_result) {
   }
 }
 
+void NCCLContext::ShutDown(){
+  for(auto it = nccl_comms.begin(); it != nccl_comms.end(); ++it){
+    ncclCommDestroy(it->second);
+  }
```

Comment:
Should we also clear the map?  `NCCLAllreduce::InitNCCLComm()` will only initialize a communicator if it is not in the map.